### ISSUE TITLE
TypeScript function overloading causes false positive

### ISSUE THREAD

**Issue by bobaaaaa** [ArnaudBarre/eslint-plugin-react-refresh#105]
## Info

Since `0.5.0` the linting rule detected ts function overloads as an error.

## Reproducible code

```ts
import type { PropsWithChildren, ReactNode } from 'react';

export function Button(props: PropsWithChildren<{ onClick: () => void }>): ReactNode;
export function Button(props: PropsWithChildren): ReactNode {
  return <button {...props}>{props.children}</button>;
}
```

## See

https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

**Comment by ArnaudBarre** [comment]
Fixed in 0.5.1!
